operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/KubernetesDependentResource.java

```diff
@@ -75,45 +86,80 @@ public void configureWith(ConfigurationService configurationService,
       boolean addOwnerReference) {
     this.informerEventSource = informerEventSource;
     this.addOwnerReference = addOwnerReference;
-    initResourceMatcherAndUpdatePreProcessorIfNotSet(configurationService);
   }
 
-  protected void beforeCreate(R desired, P primary) {
-    if (addOwnerReference) {
-      desired.addOwnerReference(primary);
+  private class DependentOperation {
+    private final String actionName;
+
+    public DependentOperation(String actionName) {
+      this.actionName = actionName;
+    }
+
+    @SuppressWarnings("unchecked")
+    protected NonNamespaceOperation<R, KubernetesResourceList<R>, Resource<R>> prepare(R desired,
+        P primary) {
+      log.debug("{} target resource with type: {}, with id: {}",
+          actionName,
+          desired.getClass(),
+          ResourceID.fromResource(desired));
+      if (addOwnerReference) {
+        desired.addOwnerReference(primary);
+      }
+      Class<R> targetClass = (Class<R>) desired.getClass();
+      return client.resources(targetClass).inNamespace(desired.getMetadata().getNamespace());
     }
   }
 
-  @Override
-  protected boolean match(R actualResource, R desiredResource, Context context) {
-    return resourceMatcher.match(actualResource, desiredResource, context);
+  private class CreateDependentOperation extends DependentOperation implements Creator<R, P> {
+
+    public CreateDependentOperation() {
+      super("Creating");
+    }
+
+    @Override
+    public void create(R target, P primary, Context context) {
+      prepare(target, primary).create(target);
+    }
   }
 
-  @SuppressWarnings("unchecked")
-  @Override
-  protected R create(R target, P primary, Context context) {
-    log.debug("Creating target resource with type: " +
-        "{}, with id: {}", target.getClass(), ResourceID.fromResource(target));
-    beforeCreate(target, primary);
-    Class<R> targetClass = (Class<R>) target.getClass();
-    return client.resources(targetClass).inNamespace(target.getMetadata().getNamespace())
-        .create(target);
+  private class UpdateDependentOperation extends DependentOperation implements Updater<R, P> {
+    private final Matcher<R> matcher;
+    private final ResourceUpdatePreProcessor<R> processor;
+
+    @SuppressWarnings("unchecked")
+    public UpdateDependentOperation() {
+      super("Updating");
+      matcher = GenericKubernetesResourceMatcher.matcherFor(resourceType());
+      processor = KubernetesDependentResource.this instanceof ResourceUpdatePreProcessor
```

Comment:
Maybe we should have a setter for this still on KDR but that can be an another PR when needed

Reply:
There actually was one but I removed it with the idea that now everything is defined in a similar fashion using traits to override the default behavior.

Reply:
The thing is that for standalone DRs, it wouldn't be necessary to make subclasses, but it is how it is. So we will se how it evolves, maybe if someone for standalone ones would not want to do subclass we can provide an "adaptor".